src/main/java/com/google/sps/placeGuide/repository/impl/DatastorePlaceGuideRepository.java

```diff
@@ -0,0 +1,129 @@
+package com.google.sps.placeGuide.repository.impl;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.sps.placeGuide.PlaceGuide;
+import com.google.appengine.api.datastore.GeoPt;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import org.jetbrains.annotations.Nullable;
+import com.google.sps.placeGuide.repository.PlaceGuideRepository;
+
+/** Class for handling place guide repository using datastore. */
+public class DatastorePlaceGuideRepository implements PlaceGuideRepository{
+
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  public static final String ENTITY_KIND = "PlaceGuide";
+  public static final String NAME_PROPERTY = "name";
+  public static final String AUDIO_KEY_PROPERTY = "audioKey";
+  public static final String CREATOR_ID_PROPERTY = "creatorId";
+  public static final String PLACE_ID_PROPERTY = "placeId";
+  public static final String IS_PUBLIC_PROPERTY = "isPublic";
+  public static final String COORD_PROPERTY = "coord";
+  public static final String DESCRIPTION_PROPERTY = "description";
+  public static final String LENGTH_PROPERTY = "length";
+  public static final String IMG_KEY_PROPERTY = "imgKey";
+  
+  @Override
+  public void savePlaceGuide(PlaceGuide placeGuide) {
+    Entity placeGuideEntity = createPlaceGuideEntity(placeGuide);
+    datastore.put(placeGuideEntity);
+  }
+
+  private Entity createPlaceGuideEntity(PlaceGuide placeGuide) {
+
+    Entity placeGuideEntity;
+    long placeGuideId = placeGuide.getId();
+    if (Long.valueOf(placeGuideId) != null) {
+      // Use the original placeGuide's ID for when user edits a place guide.
+      placeGuideEntity = new Entity(ENTITY_KIND, placeGuideId);
+    } else {
+      // Let the ID be automatically created if the place guide is new.
+      placeGuideEntity = new Entity(ENTITY_KIND);
```

Comment:
If you generate an ID for it here, maybe you should also set the ID of the placeGuide object.

Reply:
done.